### ISSUE TITLE
`pulumi refresh` "details" does not show anything

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#3350]
I expected details to show the details of the diff.  I'm fairly sure it did at some point and that this is a regression.  We should add test coverage.

```
$ pulumi refresh
Previewing refresh (dev):
     Type                 Name         Plan     Info
     pulumi:pulumi:Stack  aws770-dev            
     ├─ aws:s3:Bucket     my-bucket-2           [diff: +accelerationStatus,arn,corsRules,hostedZoneId,lifecycleRules,loggings,region,requestPayer,tag
     └─ aws:s3:Bucket     my-bucket-1           [diff: +accelerationStatus,arn,corsRules,hostedZoneId,lifecycleRules,loggings,region,requestPayer,tag
 
Resources:
    3 unchanged

Do you want to perform this refresh?
No resources will be modified as part of this refresh; just your stack's state will be. details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws770::pulumi:pulumi:Stack::aws770-dev]

Do you want to perform this refresh?
No resources will be modified as part of this refresh; just your stack's state will be. details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws770::pulumi:pulumi:Stack::aws770-dev]

Do you want to perform this refresh?
No resources will be modified as part of this refresh; just your stack's state will be. details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws770::pulumi:pulumi:Stack::aws770-dev]

Do you want to perform this refresh?
No resources will be modified as part of this refresh; just your stack's state will be. details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::aws770::pulumi:pulumi:Stack::aws770-dev]

Do you want to perform this refresh?
No resources will be modified as part of this refresh; just your stack's state will be.
  yes
> no
  details

**Comment by pgavlin** [comment]
@lukehoban IIRC we only show state/output diffs during refresh. It looks like all of those diffs are input diffs. Can you try reproducing this on resources with output changes?

**Comment by lukehoban** [comment]
How would I know whether these are input diffs or output diffs?  Why do we only show output diffs?  In this case, there definitely should have been an output diff related to `acl` which had changed - though I recall that `acl` has had other weirdness.

This doesn't feel "right" overall though.  If we show that there are diffs, it's very strange that details doesn't show details of those diffs.

**Comment by pgavlin** [comment]
> This doesn't feel "right" overall though. If we show that there are diffs, it's very strange that details doesn't show details of those diffs.

I agree. The original display code was written before `refresh` understood how to display inputs.

This is only to clarify that I don't know that we've regressed here. I think we're still showing what we've always shown in the details view, but we've made things more confusing by showing inconsistent diffs between the summary and the details.

**Comment by Dominik-K** [comment]
@pgavlin Stumbled upon this, too. And I'm pretty sure that the details-view of `refresh` showed the diffs some months ago. So I'm certain it's a **REGRESSION**. In any case the current details-view is really useless or even disappointing.

It currently only shows lines like this:
```
[...]
No resources will be modified as part of this refresh; just your stack's state will be. details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:REDACTED::gcloud::pulumi:pulumi:Stack::gcloud-REDACTED]
        - gcp:container/nodePool:NodePool: (delete) 🔒
            [id=REDACTED/k8s-infra-c43859e]
            [urn=urn:pulumi:REDACTED::gcloud::gcp:container/cluster:Cluster$gcp:container/nodePool:NodePool::k8s-infra]
            [provider=urn:pulumi:REDACTED::gcloud::pulumi:providers:gcp::production::58198140-70df-4ed5-81d2-4b898413df9a]
        ~ gcp:storage/bucketIAMMember:BucketIAMMember: (update) 🔒
            [id=eu.artifacts.REDACTED.appspot.com/roles/cloudbuild.builds.builder/serviceaccount:gke-nodes@REDACTED.iam.gserviceaccount.com]
            [urn=urn:pulumi:REDACTED::gcloud::gcp:serviceAccount/account:Account$gcp:storage/bucketIAMMember:BucketIAMMember::gke-nodes]
            [provider=urn:pulumi:REDACTED::gcloud::pulumi:providers:gcp::production::58198140-70df-4ed5-81d2-4b898413df9a]
        ~ gcp:projects/iAMMember:IAMMember: (update) 🔒
            [id=REDACTED/roles/logging.logWriter/serviceaccount:gke-nodes@REDACTED.iam.gserviceaccount.com]
            [urn=urn:pulumi:REDACTED::gcloud::gcp:serviceAccount/account:Account$gcp:projects/iAMMember:IAMMember::gke-nodes_logWriter]
            [provider=urn:pulumi:REDACTED::gcloud::pulumi:providers:gcp::production::58198140-70df-4ed5-81d2-4b898413df9a]
    ~ gcp:compute/subnetwork:Subnetwork: (update) 🔒
        [id=europe-west1/europe-west1-848c632]
        [urn=urn:pulumi:REDACTED::gcloud::gcp:compute/subnetwork:Subnetwork::europe-west1]
        [provider=urn:pulumi:REDACTED::gcloud::pulumi:providers:gcp::production::58198140-70df-4ed5-81d2-4b898413df9a]

and many more ...
```
(Pulumi version `v2.3.0`)

**Comment by lukehoban** [comment]
This is fixed now with pulumi/pulumi-terraform-bridge#330.
